### PR TITLE
Upgrade to next 13

### LINKED ISSUE #1595: Upgrade to Next 13

### PR DESCRIPTION
**Pull request by federicobadini**
closes #1595 

This pull request includes the following updates:

### Next Image

The `next/future/image` is now the default image component as per [Next.js 13 documentation](https://nextjs.org/docs/pages/building-your-application/upgrading/version-13#image-component). Some props were removed which required changes to both usages of the `next/image` component and derived components like `ResponsiveImage`. A few style fixes were needed due to markup differences.

### Next Links

The new link component doesn't require a nested anchor tag anymore. Whenever possible, the new behavior was adopted. The `legacyBehaviour` prop is available and has been used to reduce the number of changes in this PR for components that could've needed more intricate style/markup refactoring.

### Next Script

Scripts marked as `beforeInteractive` were moved to `_document.tsx` as suggested in [Next.js documentation](https://nextjs.org/docs/messages/no-before-interactive-script-outside-document).

### Next Transpile Modules

Next.js now has its own transpile option, so the `next-transpile-modules` package is no longer needed. For more information, check [here](https://github.com/martpie/next-transpile-modules/issues/280#issuecomment-1292615030).

### Sentry Edge Config

A Sentry edge config was added to address warnings about missing Sentry config files and to prepare for experiments with edge middlewares.

### SSR Failure for Next Router

A workaround was added to prevent SSR failure in Next.js 13. Refer to these discussions for more information: [issue #5650](https://github.com/algolia/instantsearch/issues/5650) and [issue #5475](https://github.com/algolia/instantsearch/issues/5475#issuecomment-1425897322).

### MSW Issue

The MSW mocking utility is not fully compatible with Next 13, as discussed [here](https://github.com/mswjs/msw/issues/1610#issuecomment-1537208102). Since MSW will likely be replaced by other mocking strategies, problematic tests have been temporarily skipped.
The reasons we suggest prioritizing an upgrade over exploring new testing mechanisms are that Next.js 13 introduces a significant change with server components. 
Delaying the adoption of this paradigm might be disadvantageous. During these weeks, several core aspects are being discussed, such as translations management, performance improvement, and A/B testing—all of which could benefit from the enhanced functionality offered by the new framework.
For additional details see [here](https://github.com/iFixit/react-commerce/pull/1726#discussion_r1232395722)

### Minor Updates

- A missing hook dependency was added to `InstantSearchProvider.tsx`.

### QA

1. Visit Vercel preview
2. Verify functionality across the various page types 
3. Take into account that several playwright test were skipped since MSW is not yet compatible with Next.js 13 - so there might be a few more scenarios to verify manually.

### CODE REVIEW

**Comment by batbattur** [comment]
@federicobadini we merged https://github.com/iFixit/react-commerce/pull/1755 to unblock the failing tests.

File: frontend/components/sections/QuoteSection.tsx [hunk 1]
```diff
@@ -41,8 +41,10 @@ export function QuoteSection({ id, quote, author, image }: QuoteSectionProps) {
                <ResponsiveImage
                   src={image.url}
                   alt=""
-                  layout="fill"
-                  objectFit="cover"
+                  fill
```
Comment (masonmcelvain) [inline frontend/components/sections/QuoteSection.tsx:44]: I see that in some places we include `sizes="100vw"` with `fill`, and in others we don't.

The [docs](https://nextjs.org/docs/pages/api-reference/components/image#sizes) say

> If you don't specify a sizes value in an image with the fill property, a default value of 100vw (full screen width) is used.

Is the motivation for including `sizes="100vw"` to be explicit about the value of this property?

File: frontend/sentry.edge.config.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import * as Sentry from '@sentry/nextjs';
+
+const SENTRY_DSN = process.env.NEXT_PUBLIC_SENTRY_DSN;
+
+Sentry.init({
+   dsn: SENTRY_DSN,
```
Comment (masonmcelvain) [inline frontend/sentry.edge.config.ts:6]: I don't think we're using vercel edge, so this config won't be used. However, I understand we might have added this to silence a sentry warning during the build process.

If we're going to add this file, I'd like us to add this tag. If we ever start using vercel edge, the config will be ready to go.

```suggestion
   dsn: SENTRY_DSN,
      initialScope: {
      tags: {
         'next.runtime': 'edge',
      },
   },
```

**Review by masonmcelvain — DISMISSED** [review]
CR ✌🏻 Thanks for taking this on!

**Comment by k0rvusk0r4x** [comment]
Everything appears to be working as before.

QA 😺

File: frontend/tests/playwright/product/product-variant.spec.ts [hunk 3]
```diff
@@ -56,7 +56,7 @@ test.describe('Product Variant Tests', () => {
       ).not.toBeVisible();
    });
 
-   test('Switch Variants and Add To Cart', async ({
```
Comment (danielbeardsley) [inline frontend/tests/playwright/product/product-variant.spec.ts:59]: Is this meant to still be in here? This seems to skip the test that ensures variant switching and add-to-cart remain working.

Comment (danielbeardsley) [reply frontend/tests/playwright/product/product-variant.spec.ts:59]: I see the pull description now. While I understand the motivation, I'm not sure I'm comfortable deploying a major change to our platform and also disabling our most important tests. Is there someplace we need to go badly enough that we feel it's worth the risk of driving without our seatbelts?

**Comment by danielbeardsley** [comment]
There aren't any commit messages here, so it's hard to understand the purpose behind some of the changes. Can you explain some of the patterns? Like usage of `style=...` instead of attributeProps for instance.

dev_block :+1:  on us being sure this is time-critical enough to warrant disabling all our tests.

File: frontend/components/sections/QuoteSection.tsx (hunk 1 above)
Comment (federicobadini) [reply frontend/components/sections/QuoteSection.tsx:44]: The image migration script offered by Next was not running smoothly on our project. More in detail some folders were skipped or errored out, so many parts were migrated manually. These explicit properties are probably due to the fact that they were made explicit by the script. On the contrary other sections that were handled manually do not show this property when it is unnecessary.
I'm going to remove that.

**Comment by federicobadini** [comment]
I squashed the commits in order to avoid cluttering the PR with temporary changes that were made to get tests working or resolve compilation errors, which might have been confusing for the reviewers. In the future, if you prefer having the whole set of commits, I can certainly leave them 😊

Regarding the usage of `style=...` instead of proper props, the reason is that some of those props were removed in Next.js v13, such as layout or objectFit. You can find more information in the Version History section of the Next.js documentation [here](https://nextjs.org/docs/pages/api-reference/components/image#version-history).

File: frontend/tests/playwright/product/product-variant.spec.ts (hunk 3 above)
Comment (ardelato) [reply frontend/tests/playwright/product/product-variant.spec.ts:59]: Building upon what @federicobadini has mentioned, @iFixit/qae has discussed the future of `MSW` in the monorepo and are leaning towards removing it. The original motive for integrating `MSW` was to mock `server-side` requests for fetching specific product types. Nonetheless, in the monorepo we can make use of our `PHP` testing extensions to set up these products in the test DB, thereby negating the necessity of `MSW`.

To address the lack of test coverage, we are working on expanding a script that would help populate the database for manual QA. https://github.com/iFixit/ifixit/issues/48312

**Comment by danielbeardsley** [comment]
>  temporary changes that were made to get tests working or resolve compilation errors, which might have been confusing for the reviewers. In the future, if you prefer having the whole set of commits, I can certainly leave them

I understand, back and forth changes or experimentation in commit history usually aren't important and often do lead to confusion, so squashing some of that makes sense. That said, having no commit messages, no pull description, and no issue description for a major version upgrade of the platform that also disables all the tests is way outside the norms.

I'm not doubting that there are good reasons to plow ahead anyway, but I think it's a good idea to document the motivations and justifications in the issue or the pull request. Acknowledging that costs and risks were evaluated, and then recording the resulting decision is prudent and good practice and leaves a paper-trail to examine later.

I vote for filling out the pull description with some of the information from your explanatory comment above.

**Comment by federicobadini** [comment]
I have updated the PR description with many details describing what this PR is doing and why we are proposing to give precedence to this wrt waiting to solve MSW/test related issues 🙂

**Review by danielbeardsley — APPROVED** [review]
CR :+1: 

Good work!

**Comment by ACPK** [comment]
Is there any progress on accepting this PR?

**Comment by danielbeardsley** [comment]
un_dev_block :+1: I'm OK with going forward on this. Still needs QA though.

**Comment by k0rvusk0r4x** [comment]
Continues to work as expected.

QA 😺

**Pull request merged**